src/lib/dnssd/minimal_mdns/Server.cpp

```diff
@@ -367,21 +381,20 @@ CHIP_ERROR ServerBase::BroadcastImpl(chip::System::PacketBufferHandle && data, u
         /// for sending via `CloneData`
         ///
         /// TODO: this wastes one copy of the data and that could be optimized away
-        if (info->addressType == chip::Inet::IPAddressType::kIPv6)
+        if (info->mAddressType == chip::Inet::IPAddressType::kIPv6)
         {
             err = udp->SendTo(mIpv6BroadcastAddress, port, data.CloneData(), udp->GetBoundInterface());
         }
 #if INET_CONFIG_ENABLE_IPV4
-        else if (info->addressType == chip::Inet::IPAddressType::kIPv4)
+        else if (info->mAddressType == chip::Inet::IPAddressType::kIPv4)
         {
             err = udp->SendTo(mIpv4BroadcastAddress, port, data.CloneData(), udp->GetBoundInterface());
         }
 #endif
         else
         {
-            // This is a general error of internal consistency: every address has a known type
-            // Fail completely otherwise.
-            return CHIP_ERROR_INCORRECT_STATE;
+            // This is a general error of internal consistency: every address has a known type Fail completely otherwise.
+            VerifyOrDie(false);
```

Comment:
This is a  change from previous behaviour: we did not crash but rather returned an error code. The new code crashes.

Reply:
Fixed, will return CHIP_ERROR_INCORRECT_STATE now.